src/chip.tsx

```diff
@@ -67,38 +83,61 @@ const Chip: React.FC<ChipProps> = (props: ChipProps) => {
                     <Icon color="currentColor" size={pxToRem(16)} />
                 </Box>
             )}
-            <Text2 id={id} medium truncate={1} color="currentColor">
-                {children}
-            </Text2>
+            <Box paddingRight={onClose || badge ? 8 : 0}>
+                <Text2 id={id} medium truncate={1} color="currentColor">
+                    {children}
+                </Text2>
+            </Box>
         </>
     );
 
-    const paddingLeft = Icon ? 8 : 12;
-
+    if (badge) {
+        return (
+            <Box
+                className={
+                    overAlternative ? styles.chipVariants.overAlternative : styles.chipVariants.default
+                }
+                paddingLeft={paddingLeft}
+                paddingRight={paddingIcon}
+                {...getPrefixedDataAttributes(dataAttributes, 'Chip')}
+            >
+                {body}
+                <div
+                    style={{
+                        display: 'flex',
+                        justifyContent: 'center',
+                        alignItems: 'center',
+                        width: badgeSize,
+                    }}
+                >
+                    {renderBadge()}
+                </div>
+            </Box>
+        );
+    }
     if (onClose) {
         return (
             <Box
                 className={
                     overAlternative ? styles.chipVariants.overAlternative : styles.chipVariants.default
                 }
                 paddingLeft={paddingLeft}
+                paddingRight={paddingIcon}
                 {...getPrefixedDataAttributes(dataAttributes, 'Chip')}
             >
                 {body}
-                <Box paddingLeft={4}>
-                    <IconButton
-                        size={24}
-                        style={{
-                            display: 'flex',
-                            justifyContent: 'center',
-                            alignItems: 'center',
-                        }}
-                        aria-label={texts.closeButtonLabel}
-                        onPress={() => onClose()}
-                    >
-                        <IconCloseRegular size={16} color={vars.colors.neutralMedium} />
-                    </IconButton>
-                </Box>
+                <IconButton
+                    size={24}
```

Comment:
This is impacting in spacing between label and "X" icon. Can we change the space between "X" and label from 8 to 4 when onclose icon is presented and keep it 8 for badge

Is it possible? 
![image](https://github.com/Telefonica/mistica-web/assets/6722153/2bb72871-fa32-43ec-813e-0ac107dd7b3e)

I'm trying to adapt as possible to specs measures keeping 24px for bounding box (iconButton) instead of 16.

The easy solution is changing IconButton size to 16

@atabel wdyt?

Reply:
I wouldn't reduce the IconButton size, we need a big enough tappeable area